### DISCUSSION TITLE
how to invalidate all cache?

### DISCUSSION THREAD

**Discussion by peterweb2005** [rrousselGit/riverpod#1991] [General]
thanks

case: 
my app need always update (may be in interval), as other users may updated their created content
so, i need to create a timer hook, and on event invalidate all cache,

so, how to invalidate all cache?

actually, my current situation, every request has a time argument,
when i update the time argument (with provider then notify), the content with attached provider will refresh

thanks

**Comment by rrousselGit** [comment]
Invalidating "all" the cache is not something you truly want to do.

If you want to update a provider every few seconds, you can do:

```dart
FutureProvider((ref) {
  final timer = Timer(duration, (_) => ref.invalidateSelf());
  ref.onDispose(timer.cancel);
  
  return fetchWhatever()
});
```

**Comment by peterweb2005** [comment]
can it invalidate without refresh?

my currently protential solution, update all timestamps,
as all my providers (of diff data types) family parameterized with timestamp argument
